### ISSUE TITLE
Special remote error upon `drop` and subsequent `get` of content

### ISSUE THREAD

**Issue by m-wierzba** [christian-monch/datalad-compute#5]
I have tested the example described in the README. I get the follwing error upon `drop` and subsequent `get` of the content:

```
❯ datalad drop name-1.txt
drop(ok): name-1.txt (file)

❯ cat name-1.txt         
cat: name-1.txt: No such file or directory

❯ datalad get name-1.txt
get(error): name-1.txt (file) [external special remote error: CommandError: 'git -c diff.ignoreSubmodules=none -c core.quotepath=false annex find --anything --json --json-error-messages -c annex.dotfiles=true -- /private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmpdz24o0n2/name-2.txt' failed with exitcode 1 under /private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmpdz24o0n2 [info keys: stdout_json] [err: 'fatal: /private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmpdz24o0n2/name-2.txt: '/private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmpdz24o0n2/name-2.txt' is outside repository at '/Users/cynamon/Downloads/compute-test-4'
external special remote error: CommandError: 'git -c diff.ignoreSubmodules=none -c core.quotepath=false annex find --anything --json --json-error-messages -c annex.dotfiles=true -- /private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmp2z_fhe6o/name-2.txt' failed with exitcode 1 under /private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmp2z_fhe6o [info keys: stdout_json] [err: 'fatal: /private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmp2z_fhe6o/name-2.txt: '/private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmp2z_fhe6o/name-2.txt' is outside repository at '/Users/cynamon/Downloads/compute-test-4'
external special remote error: CommandError: 'git -c diff.ignoreSubmodules=none -c core.quotepath=false annex find --anything --json --json-error-messages -c annex.dotfiles=true -- /private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmpwoq_p72f/name-2.txt' failed with exitcode 1 under /private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmpwoq_p72f [info keys: stdout_json] [err: 'fatal: /private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmpwoq_p72f/name-2.txt: '/private/var/folders/xg/98hhjjns1y75xykkjj_k11km0000gn/T/tmpwoq_p72f/name-2.txt' is outside repository at '/Users/cynamon/Downloads/compute-test-4']
```

I'm on MacOS, my environment is as follows:

```
❯ datalad --version  
datalad 1.1.1+3.g6577e72d7

❯ git-annex version  
git-annex version: 10.20240808
```

I think I'm using the most recent version of `datalad-compute`, but just in case:

```
❯ cd datalad-compute
❯ git log --oneline -1                                                           
2b3d030 (HEAD -> main, origin/main, origin/HEAD) Merge pull request #4 from christian-monch/io-lists
```

**Comment by christian-monch** [comment]
Thanks for the report. I can reproduce the error here. I'll look into this and fix it now.

**Comment by christian-monch** [comment]
@m-wierzba: I think I fixed the problem. It seems that `Dataset.unlock` does not use the path of  `Dataset` to determine what to unlock, but the current working directory. I pushed a workaround.

The example does now work in my environment. If you can verify that it runs for you as well, we can close this issue.

**Comment by m-wierzba** [comment]
Thank you, I can confirm that the problem is gone!

**Comment by m-wierzba** [comment]
Resolved by 635854e16af0900fdefaf34b3624aee4e6cca0ac, closing the issue now!
